pypodio2/transport.py

```diff
@@ -11,6 +14,41 @@
 import json
 
 
+class _retry:
+    """Exponential backoff decorator for TransportExceptions."""
+
+    def __init__(self, retries, delay, backoff):
+        self.retries = retries
+        self.delay = delay
+        self.backoff = backoff
+
+    def __call__(self, func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            for try_num in range(self.retries):
+                try:
+                    return func(*args, **kwargs)
+                except TransportException as exc:
+                    status = exc.status.status
+                    is_last_retry = try_num == self.retries - 1
+                    if is_last_retry:
+                        raise
+                    elif status >= 500:
+                        time.sleep(self._calc_delay(try_num))
+                        continue
+                    else:
+                        raise
+
+        return wrapper if self._retry_enabled else func
+
+    def _calc_delay(self, try_num):
+        return self.delay * pow(self.backoff, try_num)
```

Comment:
I'd suggest a add a bit of jitter here, just as a good practice. Google's [SRE book](https://landing.google.com/sre/sre-book/chapters/addressing-cascading-failures/) has some interesting background to that topic.

Reply:
...then again, this will require the tests to be modified I just realized. Consider the suggestion optional then.

Reply:
Thanks for the link, will change.